src/services/AttachmentResolver.js

```diff
@@ -99,6 +138,40 @@ export default class ImageResolver {
 		})
 	}
 
+	#getMediaPreviewUrl(mediaFileName) {
+		if (this.#user || !this.#shareToken) {
+			return generateUrl('/apps/text/mediaPreview?documentId={documentId}&sessionId={sessionId}&sessionToken={sessionToken}&mediaFileName={mediaFileName}', {
+				...this.#textApiParams(),
+				mediaFileName,
+			})
+		}
+
+		return generateUrl('/apps/text/mediaPreview?documentId={documentId}&sessionId={sessionId}&sessionToken={sessionToken}&mediaFileName={mediaFileName}&shareToken={shareToken}', {
+			...this.#textApiParams(),
+			mediaFileName,
+			shareToken: this.#shareToken,
+		})
+	}
+
+	#getMediaMetadataUrl(mediaFileName) {
+		if (this.#user || !this.#shareToken) {
+			return generateUrl('/apps/text/mediaMetadata?documentId={documentId}&sessionId={sessionId}&sessionToken={sessionToken}&mediaFileName={mediaFileName}', {
+				...this.#textApiParams(),
+				mediaFileName,
+			})
+		}
+
+		return generateUrl('/apps/text/mediaMetadata?documentId={documentId}&sessionId={sessionId}&sessionToken={sessionToken}&mediaFileName={mediaFileName}&shareToken={shareToken}', {
+			...this.#textApiParams(),
+			mediaFileName,
+			shareToken: this.#shareToken,
+		})
+	}
+
+	getMediaMetadata(mediaFileName) {
+		return axios.get(this.#getMediaMetadataUrl(mediaFileName))
+	}
```

Comment:
Could we maybe move the call to `axios` to the calling function? So far this class just encapsulates the logic of generating the urls. Without the axios call it has no side effects - which makes it very easy to test and reason about.

That might also make it easy to add a test for the logic generating the url.

Reply:
I completely agree. Done.

Metadata url generation is now tested.

Reply:
thanks